### PR TITLE
[WIP] perf: reduce bundle size

### PR DESCRIPTION
**Pull request by hugomrdias**
Before/After:
modules        1549	        946         -38.92%
unminified     8 617 735	3 629 020	-57.88893485%
minified       3 007 790	1 439 307	-52.14735736%
minified+gzip  773.59	    386.56	    -50.03037785%

Analyzer       
https://bundle-size.netlify.com/ipfs-before.html

screenshot 2018-12-24 at 12 16 39

https://bundle-size.netlify.com/ipfs-after.html 

screenshot 2018-12-24 at 12 17 05

### CODE REVIEW

File: test/http-api/inject/pin.js [hunk 1]
```diff
@@ -115,7 +115,7 @@ module.exports = (http) => {
       })
     })
 
-    describe('ls', () => {
+    describe.only('ls', () => {
```
Comment (achingbrain) [inline test/http-api/inject/pin.js]: Remove pls

**Comment by alanshaw** [comment]
@hugomrdias are you still tracking this? Can you list out the dependent PRs in the description above so I can help you push them along?

**Comment by hugomrdias** [comment]
waiting on mplex we are getting there just a little bit longer

File: src/http/api/resources/config.js [hunk 2]
```diff
@@ -1,10 +1,10 @@
 'use strict'
 
 const debug = require('debug')
-const get = require('lodash/get')
-const set = require('lodash/set')
-const log = debug('ipfs:http-api:config')
-log.error = debug('ipfs:http-api:config:error')
+const get = require('dlv')
+const set = require('just-safe-set')
+const log = debug('jsipfs:http-api:config')
```
Comment (alanshaw) [inline src/http/api/resources/config.js]: These have been [standardized to all start with "ipfs:"](https://github.com/ipfs/js-ipfs/pull/1911).

```suggestion
const log = debug('ipfs:http-api:config')
```

File: src/http/api/resources/config.js [hunk 3]
```diff
@@ -1,10 +1,10 @@
...
+log.error = debug('jsipfs:http-api:config:error')
```
Comment (alanshaw) [inline src/http/api/resources/config.js]: These have been [standardized to all start with "ipfs:"](https://github.com/ipfs/js-ipfs/pull/1911).

```suggestion
log.error = debug('ipfs:http-api:config:error')
```

File: src/core/runtime/ipld-browser.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+'use strict'
+const mergeOptions = require('merge-options')
+
+module.exports = (blockService, options = {}) => {
+  return mergeOptions({
+    blockService: blockService
+  }, options)
+}
```
Comment (alanshaw) [inline src/core/runtime/ipld-browser.js]: Until we have multiple browser builds we need to include bitcoin, ethereum, git and zcash.

**Comment by alanshaw** [comment]
Shall we close this one @hugomrdias?

**Pull request closed without merging**